NewForce-at-Mountwest/bangazon-api-farfalle
# Get, Post, and Put for Departments + tests

# Description
I have created a controller for the Department resource that supports 'get all', 'get single', post, and put.  
'Get single' and 'get all' support a query string of ''''?_include=employees''''
'Get all' also supports a ```?_filter=budget&_gt=30000``` to look at all the departments with a budget greater than or equal to 30000.

## Related Ticket(s)
[7](https://trello.com/c/bLd0HcCt/7-ticket-7-allow-developers-to-access-the-department-resource)


## Steps to Test
Be sure to add and commit on your own branch before checking out to master!
git fetch --all to get access to my branch

git checkout sw-departments-controller 
git pull origin sw-departments-controller

type start BangazonAPI.sln if you don't already have it open

In the Solution Explorer, right click on 'TestBangazonApi' and select 'Run Tests'.  These test all the base methods as well as get methods that include query strings.

Because this controller only requires get, post, and put, I created a secret delete method using the query string q=delete_test_item.  To ensure that it ACTUALLY deleted, you can either check in the browser or postman after running the program. ````https://localhost:5001/api/department```


You should NOT see anything that says "test" or "cool" in departments. You should only see the departments already seeded in your database (we started with 3). If you need seed data for your database, it can be found [here.](https://github.com/NewForce-at-Mountwest/bangazon-api-farfalle/blob/master/Bangazon_boilerplate_db.sql)

# Checklist:
